src/functions/searchResults/framework/__tests__/handler.spec.ts

```diff
@@ -0,0 +1,92 @@
+      },
+    });
+
+    dummyContext = lambdaTestUtils.mockContextCreator(() => null);
+    process.env.EMPLOYEE_ID_EXT_KEY = 'extn.employeeId';
+
+    spyOn(searchResultsSvc, 'getConciseSearchResults').and.callFake(moqSearchResults.object);
+    spyOn(configSvc, 'bootstrapConfig').and.callFake(moqBootstrapConfig.object);
+  });
+
+  describe('configuration initialisation', () => {
+    it('should always bootstrap the config', async () => {
+      await handler(dummyApigwEvent, dummyContext);
+      moqBootstrapConfig.verify(x => x(), Times.once());
+    });
+  });
+
+  describe('handling of no parameters', () => {
+    it('should fail with bad request and give an error message', async () => {
+      dummyApigwEvent.queryStringParameters = null;
+      const resp = await handler(dummyApigwEvent, dummyContext);
+      expect(resp.statusCode).toBe(400);
+      expect(JSON.parse(resp.body)).toBe('Query parameters have to be supplied');
+    });
+  });
+
+  describe('handling of only isLDTM parameter', () => {
+    it('should fail with bad request and give an error message', async () => {
+      dummyApigwEvent.queryStringParameters['isLDTM'] = 'true';
+      const resp = await handler(dummyApigwEvent, dummyContext);
+      expect(resp.statusCode).toBe(400);
+      expect(JSON.parse(resp.body)).toBe('Query parameters have to be supplied');
+    });
+  });
+
+  describe('using invalid query parameters', () => {
+    it('should fail with bad request and give an error message', async () => {
+      dummyApigwEvent.queryStringParameters['whatever'] = 'randomvalue';
+      const resp = await handler(dummyApigwEvent, dummyContext);
+      expect(resp.statusCode).toBe(400);
+      expect(JSON.parse(resp.body)).toBe('Query parameters have to be supplied');
+    });
+  });
+
+  describe('using invalid query parameters', () => {
+    it('should fail with bad request and give an error message', async () => {
+      dummyApigwEvent.queryStringParameters['whatever'] = 'randomvalue';
+      const resp = await handler(dummyApigwEvent, dummyContext);
+      expect(resp.statusCode).toBe(400);
+      expect(JSON.parse(resp.body)).toBe('Query parameters have to be supplied');
+    });
+  });
+
+  describe('using valid query parameters as LDTM', () => {
+    it('gets the relevant results', async () => {
+      dummyApigwEvent.queryStringParameters['isLDTM'] = 'true';
+      dummyApigwEvent.queryStringParameters['startDate'] = queryParameter.startDate;
+      dummyApigwEvent.queryStringParameters['endDate'] = queryParameter.endDate;
+      const testResult : TestResultRecord = {
+        test_result: 'Nothing',
+      };
+      const fakeTestResult = Mock.ofType<Promise<TestResultRecord>>();
+      fakeTestResult.setup((x: any) => x.then).returns(() => testResult);
+      moqSearchResults.setup(x => x(It.isAny())).returns(() => Promise.resolve(fakeTestResult.object));
+      const resp = await handler(dummyApigwEvent, dummyContext);
```

Comment:
Even though I'm mocking the searchResults service and returning a TestResultRecord, it seems to returns undefined, any ideas?